main.m

```diff
@@ -41,8 +56,64 @@ void printToStdOut(NSString *format, ...) {
 	[formattedString release];
 }
 
-void printUsage() {
-	printToStdOut(@"Usage: keychain_dumper [-e]|[-h]|[-agnick]\n");
+NSString *runProcess(NSString *executablePath, NSArray *args)
+{
+    NSPipe *pipe = [NSPipe pipe];
+    NSFileHandle *file = pipe.fileHandleForReading;
+
+    NSTask *task = [[NSTask alloc] init];
+    task.launchPath = executablePath;
+    task.arguments = args;
+    task.standardOutput = pipe;
+    task.standardError = [NSPipe pipe];
+    [task launch];
+
+    NSData *data = [file readDataToEndOfFile];
+    [file closeFile];
+
+    return [[NSString alloc] initWithData: data encoding: NSUTF8StringEncoding];
+}
+
+NSString *saveDataTemporarily(NSData *data)
+{
+    NSString *tmpPath = @"/tmp/data";
+    [data writeToFile:tmpPath atomically:YES];
+    return tmpPath;
+}
+
+NSString *runOpenSSLWithArgs(NSArray *args)
```

Comment:
As noted..I've fallen out of the iOS world. But..it seems this code implies iOS still has nothing for parsing keys sensibly? Would it be horrible to vendor in some kind of openssl framework to not have to shell out?

Reply:
This request is fair enough, I will take a look at this once I got some time.

Reply:
As far as I'm concerned there are OpenSSL frameworks in the wild (e.g. https://github.com/krzyzanowskim/OpenSSL). Using them in this program, might lead to portability issues, since one must first download/build this framework for building the binaries. The actual approach is to use OpenSSL provided by Cydia. What do you think about this?

Reply:
That would result in the binary being dynamically linked, right? That would sill require some hoops to jump through to get it building on the Mac and such, no? All things being equal I think it would be preferable to statically compile/link things such that the binary can be dropped onto a phone with zero expectations about the environment. I've not used cocoapods before, but so long as it is seems sensibly supported means of getting openssl into the project I'm open to using it.

Reply:
Would it make any sense to open a PR that just updates the bits that don't rely on shelling out to openssl and then tackling the question of how to best handle shelling out or linking against openssl in a standalone PR?

Reply:
I don't see the benefit. In my eyes, everyone who uses this tool, can also install openssl with cydia. (if it's not installed beforehand) 

To be honest it's too much of a hurdle for me to implement it directly in the binary. We could modify the code in order to skip checks that depend on openssl, if openssl is not installed. I personally prefer the actual setup, but your opinion may differ. :)

Reply:
That is reasonable. I had a _preference_, but if it is too tricky to do easily, that is fine. Let me give this a once-over next week, see if I can get it all to build, and give it a review. Thanks again!

Reply:
Your welcome, how is the actual state?
Regards